ast_test.go

```diff
@@ -1266,6 +1267,50 @@ func TestEvalType(t *testing.T) {
 	}
 }
 
+func Benchmark_NumberLiteral_String(b *testing.B) {
+	// Generate test numbers using an exponential distribution.
+	// The numbers are heavily weighted towards fractional numbers, followed by a whole numbers.
+	// Nan, +Inf, and -Inf round out the test data at much lower percentages.
+	tc := struct {
+		Fractional int
+		Whole      int
+		NaN        int
+		PosInf     int
+		NegInf     int
+	}{}
+	nl := make([]influxql.NumberLiteral, b.N, b.N)
+	for i := range nl {
+		var n float64
+		sel := rand.ExpFloat64() * 10
+		switch {
+		case sel >= 12.5 || sel <= 5.0:
+			n = rand.Float64()
+			tc.Fractional++
+		case sel > 12.0:
+			n = math.Inf(-1)
+			tc.NegInf++
+		case sel > 11.5:
+			n = math.Inf(1)
+			tc.PosInf++
+		case sel > 11.0:
+			n = math.NaN()
+			tc.NaN++
+		case sel > 5.0:
+			n = float64(rand.Int31())
+			tc.Whole++
+		default:
+			panic("Shouldn't get here")
+		}
+		nl[i] = influxql.NumberLiteral{Val: n}
+	}
+	fmt.Printf("tc: %+v\n", tc)
+	b.ResetTimer()
+
```

Comment:
how does this benchmark work without respecting `b.N`?

Reply:
It does respect `b.N`, although I admit it is somewhat hidden. `nl` is created to be of length `b.N`. All loops after that are `for i := range nl`, which is equivalent to `for i := 0; i < b.N, i++`.

Reply:
finally, last question:  why use the two value range but use the empty assigment operator?  why not simply:

```
for i := range nl {
  ...
}
```

Again, not that i'm advocating using ranging over `nl` as a reasonable alternative to the classic `for i := 0; i < b.N; i++` pattern.

It is an interesting choice.